pkg/orchestrator/provider/docker/client.go

```diff
@@ -385,6 +386,11 @@ func (c *Client) getNetworkIDFromName(ctx context.Context, networkName string) (
 		return "", fmt.Errorf("scan network not found: %w", err)
 	}
 	if len(networks) > 1 {
+		for _, n := range networks {
+			if n.Name == networkName {
+				return n.ID, nil
+			}
+		}
```

Comment:
why is this needed if the `networkName` is used in the filter?

Reply:
Thanks for reviewing, @FrimIdan! This is needed because when I was testing `NetworkList` returns both the `vmclarity-e2e` and `vmclarity-e2e-asset` networks, i.e. the filter searches by substring not full string. Therefore, we need to check the full string match here and return the corresponding ID.